File: archive/frontend/src/types/conversation.ts
export interface Node {
  id: string
  user_msg: string | null
  ai_msg: string | null
  parent_id: string | null
  annotations: string
  timestamp: string
  created_at: string
  updated_at: string
}

export interface Branch {
  id: string
  name: string
  conversation_id: string
  root_node_id: string
  node_ids: string[]
  created_at: string
  updated_at: string
}

export interface Conversation {
  id: string
  title: string
  created_at: string
  updated_at: string
  current_branch_id: string | null
  current_node_id: string | null
}

export interface CreateConversationRequest {
  title: string
}

export interface CreateBranchRequest {
  name: string
  from_node_id: string
}

export interface AddMessageRequest {
  user_msg: string
  parent_id: string
}

export interface AIResponseRequest {
  messages: Array<{
    role: 'user' | 'assistant'
    content: string
  }>
  model?: string
}

export interface AIResponse {
  content: string
  model: string
}

export interface ConversationTree {
  conversation: Conversation
  branches: Branch[]
  nodes: Node[]
  current_branch: Branch | null
  current_node: Node | null
}

File: archive/backend/schemas/conversation.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

# Base schemas
class NodeBase(BaseModel):
    user_msg: Optional[str] = None
    ai_msg: Optional[str] = None
    parent_id: Optional[str] = None
    annotations: Optional[str] = ""

class NodeCreate(NodeBase):
    pass

class NodeUpdate(BaseModel):
    user_msg: Optional[str] = None
    ai_msg: Optional[str] = None
    annotations: Optional[str] = None

class Node(NodeBase):
    id: str
    conversation_id: str
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

class BranchBase(BaseModel):
    name: str
    root_node_id: str

class BranchCreate(BranchBase):
    pass

class Branch(BranchBase):
    id: str
    conversation_id: str
    node_ids: List[str]
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

class ConversationBase(BaseModel):
    title: str

class ConversationCreate(ConversationBase):
    pass

class ConversationUpdate(BaseModel):
    title: Optional[str] = None
    current_branch_id: Optional[str] = None
    current_node_id: Optional[str] = None

class Conversation(ConversationBase):
    id: str
    created_at: datetime
    updated_at: datetime
    current_branch_id: Optional[str] = None
    current_node_id: Optional[str] = None
    
    class Config:
        from_attributes = True

# Complex schemas
class ConversationTree(BaseModel):
    conversation: Conversation
    branches: List[Branch]
    nodes: List[Node]
    current_branch: Optional[Branch] = None
    current_node: Optional[Node] = None

# Request schemas
class CreateBranchRequest(BaseModel):
    name: str
    from_node_id: str

class AddMessageRequest(BaseModel):
    user_msg: str
    parent_id: str

class AIResponseRequest(BaseModel):
    messages: List[dict]
    model: Optional[str] = "gpt-4"

class AIResponse(BaseModel):
    content: str
    model: str